.travis.yml

```diff
@@ -70,65 +259,160 @@ jobs:
-      - libpaper-utils
-      - libperl-critic-perl
-      - libumfpack5.6.2
-      - libxml2-utils
-      - luarocks
-      - mercurial
-      # menhir
-      - mono-mcs
-      # ocaml
-      # opam
-      - php-codesniffer
-      - r-base
-      - verilator
+    # Entries generates from `supported_versions`
+    - stage: sentinel
+      language: python
+      python: 3.6
+      addons:
+        apt:
+          packages: clang-3.4
+
+    - stage: test-languages
+      language: python
+      python: 3.5
+      addons:
+        apt:
+          packages: clang-3.4
+
+    - stage: test-other-versions
+      language: python
+      python: 3.4.4
+      addons:
+        apt:
+          packages: clang-3.4
+
+    - stage: test-languages
+      language: csharp
+      mono: 5.20.1
+
+    - stage: test-languages
+      language: r
+      r: release
+
+    - stage: test-other-versions
+      language: r
+      r: devel
+
+    - stage: test-other-versions
+      language: r
+      r: oldrel
+
+    - stage: test-languages
+      language: node_js
+      node_js: '10'
+
+    - stage: test-other-versions
+      language: node_js
+      node_js: 9
+
+    - stage: test-other-versions
+      language: node_js
+      node_js: 8
+
+    - stage: test-other-versions
+      language: node_js
+      node_js: 7
+
+    - stage: test-other-versions
+      language: node_js
+      node_js: 6
+
+    - stage: test-languages
+      language: julia
+      julia: 1.1
+      env: JL_PKG=CoalaBears
+      install:
+        - julia --color=yes .ci/deps.julia.jl
+        # Verify compilation works
+        - julia -e 'import Lint.lintfile'
+
+    - stage: test-other-versions
+      language: julia
+      julia: 1.0
+      env: JL_PKG=CoalaBears
+      install:
+        - julia --color=yes .ci/deps.julia.jl
+        # Verify compilation works
+        - julia -e 'import Lint.lintfile'
+
+    - stage: test-other-versions
+      language: julia
+      julia: 0.7.0
+      env: JL_PKG=CoalaBears
+      install:
+        - julia --color=yes .ci/deps.julia.jl
+        # Verify compilation works
+        - julia -e 'import Lint.lintfile'
+
+    - stage: test-languages
+      language: dart
+      dart: 1.15.0
+
+    - stage: test-other-versions
+      language: dart
+      dart: 1.14.2
+
+    - stage: test-languages
+      language: perl
+      perl: 5.14
+
+    - stage: test-languages
+      language: go
+      go: 1.11
+      install: true
+
+    - stage: test-other-versions
+      language: go
+      go: '1.10'
+      install: true
+
+    - stage: test-languages
+      language: ruby
+      rvm: 2.4
+
+    - stage: test-other-versions
+      language: ruby
+      rvm: 2.3
+
+    - stage: test-other-versions
+      language: ruby
+      rvm: 2.2
+
+    - stage: test-other-versions
+      language: ruby
+      rvm: 2.1
+
+    - stage: test-languages
+      language: php
+      php: 7.2
+
+    - stage: test-other-versions
+      language: php
+      php: hhvm-3.18
```

Comment:
temporary breakage here https://travis-ci.community/t/php-version-hhvm-3-18-failing/4250/ - should be fixed tomorrow.  Remove this job if everything else is OK, and re-instate it later